### PR TITLE
Eq and debug traits

### PR DESCRIPTION
**Pull request by davidrusu**
Hi, thanks for this high quality crate. I ran into some missing standard trait impls when attempting to use it in my work.

Note, I've included the "developer certificate of origin" text in each commit message.

### CODE REVIEW

File: src/ppar.rs [hunk 1]
```diff
@@ -4,6 +4,7 @@ use super::*;
 use crate::{Error, Result};
 
 /// Persistent array using rope-data-structure.
+#[derive(Debug)]
```
Comment (prataprc) [inline src/ppar.rs:7]: So there is this method `pretty_print()` available on Vector and Node types. The primary intention was to be able to debug the tree while testing. I am not sure Debug will be of practical use when the Tree depth is large, and testing it with big data-set.

I am okay with this patch, just wanted to know the reason for adding this. Thanks,

**Comment by prataprc** [comment]
I am of the understanding that `git -s` adds a signed-off message to comment and implies a DCO. Correct me if I am wrong.  I think we can remove the DCO content from the comment and replace it with "Signed-off" line [like mentioned here](https://writing.kemitchell.com/2021/07/02/DCO-Not-CLA.html)

Otherwise I am fine with this commit-set. Thanks for your contribution.

**Comment by davidrusu** [comment]
I've amended the commits to replace the DCO with the sign off's. I'm not sure what the legal implications of the sign off's are, but I can tell you that I am within my rights to contribute these changes to your project. (I don't have an employer who can claim IP ownership over these changes).

**Comment by prataprc** [comment]
Thanks.

**Pull request merged**